packages/appium/lib/extension/manifest.js

```diff
@@ -0,0 +1,564 @@
+// @ts-check
+
+/**
+ * Module containing {@link Manifest} which handles reading & writing of extension config files.
+ */
+
+import {env, fs, mkdirp} from '@appium/support';
+import _ from 'lodash';
+import path from 'path';
+import YAML from 'yaml';
+import {DRIVER_TYPE, PLUGIN_TYPE} from '../constants';
+import log from '../logger';
+import {INSTALL_TYPE_NPM} from './extension-config';
+import {packageDidChange} from './package-changed';
+
+/**
+ * Default depth to search in directory tree for whatever it is we're looking for.
+ *
+ * It's 4 because smaller numbers didn't work.
+ */
+const DEFAULT_SEARCH_DEPTH = 4;
+
+/**
+ * Default options for {@link findExtensions}.
+ * @type {Readonly<import('klaw').Options>}
+ */
+const DEFAULT_FIND_EXTENSIONS_OPTS = Object.freeze({
+  depthLimit: DEFAULT_SEARCH_DEPTH,
+  /* istanbul ignore next */
+  filter: (filepath) => !path.basename(filepath).startsWith('.'),
+});
+
+/**
+ * Current configuration schema revision!
+ */
+const CONFIG_SCHEMA_REV = 2;
+
+/**
+ * @type {`${typeof DRIVER_TYPE}s`}
+ */
+const CONFIG_DATA_DRIVER_KEY = `${DRIVER_TYPE}s`;
```

Comment:
what is the magic `s` suffix?

Reply:
Well... it's an `s`.  `driver` becomes `drivers`.  This is a key in the root of `extensions.yaml`.

Reply:
maybe a small explanation comment would make it cleaner. For me this magic char looked quite cryptic